src/main/java/edu/harvard/iq/dataverse/harvest/server/web/servlet/OAIServlet.java

```diff
@@ -256,9 +259,18 @@ private void processRequest(HttpServletRequest httpServletRequest, HttpServletRe
                         "Sorry. OAI Service is disabled on this Dataverse node.");
                 return;
             }
-                        
-            RawRequest rawRequest = RequestBuilder.buildRawRequest(httpServletRequest.getParameterMap());
-            
+
+            RawRequest rawRequest = null;
+            try {
+                rawRequest = RequestBuilder.buildRawRequest(httpServletRequest.getParameterMap());
+            } catch (BadVerbException bve) {
+                // Verb.Type is required. Hard-code one.
+                rawRequest = new RawRequest(Verb.Type.Identify);
+                // Ideally, withError would accept a BadVerbException.
+                BadArgumentException bae = new BadArgumentException(bve.getLocalizedMessage());
+                rawRequest.withError(bae);
```

Comment:
On a second thought, there should be a less hacky way to handle this, when a rawRequest cannot be created, that does not rely on creating one with a hard-coded valid verb and .withError() etc. 
Note that DataProvider also has a version of the .handle() method that takes a map:
```
  public OAIPMH handle(Map<String, String[]> queryParameters) {
```
https://github.com/gdcc/xoai/blob/xoai-5.2.0/xoai-data-provider/src/main/java/io/gdcc/xoai/dataprovider/DataProvider.java

Also,
https://github.com/gdcc/xoai/blob/xoai-5.2.0/xoai-data-provider/src/test/java/io/gdcc/xoai/dataprovider/DataProviderTest.java#L49 etc.

Reply:
Awesome, I'll take a look. Thanks!

Reply:
Ok, fixed in 88af3d4ed1.

One thing I'll note is that I believe we can continue refactoring even further. I tried replacing `catch  BadVerbException` with `catch OAIException` and the tests still pass. Let me know.

Reply:
Hmm, that I would not remember off the top of my head - if it was possible to encounter any other OAIException there; but I'm guessing that anything with a valid verb, and any combination of other invalid values should be properly handled by the existing code.  
Catching `BadVerbException` solves the problem at hand, so I'm happy to approve as is.

Reply:
Make sense. Thanks for approving!